### PR TITLE
AccessibilityInfo: added missing tests for Promise based methods, align null checks and error message code style 

### PR DESCRIPTION
**Pull request by chicio**
## Summary:

This PR continue the work I'm doing to improve the `AccessibilityInfo` module (#56066 and connected PRs).

In this new PR I added the missing tests for all the Promise based methods in the `AccessibilityInfo`, to prevent regressions similar to those fixed in the previous PRs.

This PR also includes a couple of small code style consistency improvement:
- aligned the null check in `getRecommendedTimeoutMillis` to the other ones already present in the file
- aligned the error messages returned from `isGrayscaleEnabled`, `isInvertColorsEnabled` and `isReduceMotionEnabled` to the other ones
- added the docs links where was missing (related to the PR I opened on the react-native doc repo)

## Changelog:

[GENERAL] [ADDED] - AccessibilityInfo: added missing tests for Promise based methods, align null checks and error message code style 

## Test Plan:

I run the tests that I added to verify their correctness. Some of them were failing because of the change of the error messages, and I made them green by changing the message in the production code. For this others, I verified their correctness by breaking the production code to see if they were failing.

### CODE REVIEW

**Comment by chicio** [comment]
Hi @vzaidman, maybe you can help me to import also this one? Thank you so much 🙏

**Comment by chicio** [comment]
Hi @cortinico, can you help me with this PR?  Thanks :pray:

File: packages/react-native/Libraries/Components/AccessibilityInfo/__tests__/AccessibilityInfo-test.js [hunk 1]
```diff
@@ -66,9 +171,445 @@ describe('AccessibilityInfo', () => {
...
+
+    describe('iOS', () => {
+      it('should call getCurrentReduceMotionState if available', async () => {
+        /* $FlowFixMe[incompatible-type] */
+        Platform.OS = 'ios';
+
+        const isReduceMotionEnabled =
+          await AccessibilityInfo.isReduceMotionEnabled();
+
+        expect(isReduceMotionEnabled).toBe(true);
```
Comment (cortinico) [inline packages/react-native/Libraries/Components/AccessibilityInfo/__tests__/AccessibilityInfo-test.js:590]: Can you also assert this?

`expect(mockGetCurrentReduceMotionState).toHaveBeenCalled()`

File: packages/react-native/Libraries/Components/AccessibilityInfo/__tests__/AccessibilityInfo-test.js [hunk 2]
```diff
@@ -66,9 +171,445 @@ describe('AccessibilityInfo', () => {
...
+      });
+
+      it('should throw error if isReduceMotionEnabled is not available', async () => {
+        /* $FlowFixMe[incompatible-type] */
+        Platform.OS = 'android';
+
+        const nativeAccessibilityInfoModule = jest.requireMock(
+          '../NativeAccessibilityInfo',
+        );
+        nativeAccessibilityInfoModule.default = null;
```
Comment (cortinico) [inline packages/react-native/Libraries/Components/AccessibilityInfo/__tests__/AccessibilityInfo-test.js:566]: This is different from the test title: 'should throw error if isReduceMotionEnabled is not available'

You should `mockNativeAccessibilityInfoAndroid.isReduceMotionEnabled = null;`

instead

**Review by cortinico — APPROVED** [review]
Generally looks good to me, but left a couple of nits 👍

File: packages/react-native/Libraries/Components/AccessibilityInfo/__tests__/AccessibilityInfo-test.js (hunk 1 above)
Comment (chicio) [reply packages/react-native/Libraries/Components/AccessibilityInfo/__tests__/AccessibilityInfo-test.js:590]: Done 👍

File: packages/react-native/Libraries/Components/AccessibilityInfo/__tests__/AccessibilityInfo-test.js (hunk 2 above)
Comment (chicio) [reply packages/react-native/Libraries/Components/AccessibilityInfo/__tests__/AccessibilityInfo-test.js:566]: Actually the test is correct, but the title wrong. 
I fixed it to match the naming used in similar tests that is `should throw error if NativeAccessibilityInfoAndroid is not available`.

**Comment by chicio** [comment]
> Generally looks good to me, but left a couple of nits 👍

Thanks 🙏  I addressed all the comment. Let me know if you spot anything else or it is ready to be merged.

File: packages/react-native/Libraries/Components/AccessibilityInfo/__tests__/AccessibilityInfo-test.js (hunk 2 above)
Comment (cortinico) [reply packages/react-native/Libraries/Components/AccessibilityInfo/__tests__/AccessibilityInfo-test.js:566]: yeah either works
